nengo/utils/compat.py

```diff
@@ -75,12 +75,24 @@ def ensure_bytes(s):
 assert TextIO
 
 
+def as_ascii(string):
+    if not PY2 and isinstance(string, bytes):  # Python 3
+        return string.decode('ascii')
+    elif not isinstance(string, str):
+        return str(string)
+    else:
+        return string
+
+
 def is_integer(obj):
     return isinstance(obj, int_types + (np.integer,))
 
 
 def is_iterable(obj):
-    return isinstance(obj, collections.Iterable)
+    if isinstance(obj, np.ndarray):
```

Comment:
Will this actually be true for 0-d arrays? I thought that they were instances of `np.generic` but not `np.ndarray`. Weird that NumPy scalars are still iterable if they can't be iterated over!